### DISCUSSION TITLE
6-1 Add Data Source

### DISCUSSION THREAD

**Discussion by aaronbirr** [ua-gist604b-f24/syllabus#31] [General]
I got as far as Add New Vector Data Set and ran into some similar connection problems. Anyone run into this and know what to do? 
Thanks 

![Screenshot 2024-11-23 183041](https://github.com/user-attachments/assets/c1e6f7fc-7544-454a-8ca0-f6e13160fdc9)

**Comment by robertgamboa10** [comment]
I have been working on it all day and I think I figured the last few steps for this module and I had the same issue at first for that part.  For the host, put "postgis" instead of "localhost"

When you get to the step that says New Layer, first publish each layer and I think I was having issues connecting to QGIS like everyone else and instead of going to Layer-> Add PostGIS Layers, I went back to the home page on GEOServer and copied the link for WFS 2.0  and went into QGIS and pasted the URL after right clicking on WFS and click on add a new connection. Sorry, hope that helps. I am still working on it right now but that is as far as I have gotten.



<img width="1136" alt="Screenshot 2024-11-23 at 5 58 26 PM" src="https://github.com/user-attachments/assets/1374b7c1-2650-4724-afe0-b6ffb8857250">
51-a8a9-981e2a8856d0">




<img width="1136" alt="Screenshot 2024-11-23 at 5 56 41 PM" src="https://github.com/user-attachments/assets/9de1e5f5-7463-4f8f-956b-950de83337f2">

**Reply by aaronbirr** [reply]
Thanks Robert! That is helpful, I will give it a shot and see.

**Reply by travisz09** [reply]
Thank you! This is precisely where I got stuck as well. Using host 'postgis' worked!

**Reply by andreasrf** [reply]
Oh this is a really good workaround. Thank you!

**Reply by aldunn4** [reply]
None of the settings I tried for the PostGIS connection worked, but the WFS option worked. Thanks.

**Reply by aaryno** [reply]
Thank you, @robertgamboa10! The issue here (and the reason why the directions are wrong) is that the network within which geoserver is operating is its local docker network. The way docker works is that it creates a sandboxed environment for every docker container that's running. These docker containers don't know about each other - they don't share files and can't see each other on the network. However, in cases where we WANT to connect the sandboxes we can create a network, put them both on the network, and then network traffic will be able to be routed back and forth between them. In this case, we created a network named "gist604b" and we gave the postgis container a network name, "postgis". Thus, geoserver can find it on the network under that name. 

![image](https://github.com/user-attachments/assets/01489a59-ce0d-4fbd-abda-425dac985d6b)
